.github/workflows/build_and_test.yml

```diff
@@ -74,11 +74,54 @@ jobs:
         shell: bash
         id: build-tag
 
-  check-codestyle-python:
+      - name: Get buildtools tag
+        run: |
+          if [[ "$GITHUB_EVENT_NAME" == "pull_request" ]]; then
+            IMAGE_TAG=$GITHUB_RUN_ID
+          else
+            IMAGE_TAG=pinned
+          fi
+          
+          echo "image_tag=${IMAGE_TAG}" >> $GITHUB_OUTPUT
+        shell: bash
+        id: buildtools-tag
+
+  check-if-build-tools-dockerfile-changed:
     needs: [ check-permissions ]
+    runs-on: ubuntu-latest
+    outputs:
+      docker_file_changed: ${{ steps.dockerfile.outputs.docker_file_changed }}
+    steps:
+      - name: Check if Dockerfile.buildtools has changed
```

Comment:
Now I see, that we don't really need `check-if-build-tools-dockerfile-changed` job, let's move this step to the `build_and_push_docker_image.yml` workflow (then we won't need to pass `docker-file-changed` from the outside).

Reply:
yep make sense